F/FastTree/build_tarballs.jl

```diff
@@ -35,6 +47,10 @@ products = [
     ExecutableProduct("fasttree_noSSE", :fasttree_noSSE),
     ExecutableProduct("fasttreeMP", :fasttreeMP),
     ExecutableProduct("fasttreeMP_noSSE", :fasttreeMP_noSSE),
+    ExecutableProduct("fasttree_Double", :fasttree_Double),
+    ExecutableProduct("fasttree_Double_noSSE", :fasttree_Double_noSSE),
+    ExecutableProduct("fasttreeMP_Double", :fasttreeMP_Double),
+    ExecutableProduct("fasttreeMP_Double_noSSE", :fasttreeMP_Double_noSSE),
```

Comment:
This is not going to work ***if*** downstream packages expect libraries with different names, hence my suggestion in https://github.com/JuliaPackaging/Yggdrasil/issues/6895#issuecomment-1594395750.  Note the "if" before, I have no idea what downstream packages expect.